spec/features/admin/payments_spec.rb

```diff
@@ -17,9 +17,10 @@
   end
 
   context "with sensitive payment fee" do
-    let(:payment_method) { order.distributor.payment_methods.first }
-
     before do
+      payment_method = create(:payment_method, distributors: [order.distributor])
+      order.distributor.payment_methods << payment_method
```

Comment:
Is the distributors' array assignment in the factory creation actually needed if we add this line above? We have an AR association specifying this relationship :thinking:

Reply:
This worked on a rails console

```ruby
irb(main):006:0> order = FactoryBot.create(:order)
irb(main):009:0> order.distributor = FactoryBot.create(:enterprise)
irb(main):012:0> payment_method = FactoryBot.create(:payment_method)
irb(main):011:0> order.distributor.payment_methods
=> []
irb(main):012:0> payment_method = FactoryBot.create(:payment_method)
irb(main):013:0> order.distributor.payment_methods << payment_method                                                                                                                                         => [#<Spree::PaymentMethod::Check id: 1, type: "Spree::PaymentMethod::Check", name: "Check", description: nil, active: true, environment: "test", created_at: "2018-11-14 10:56:13", updated_at: "2018-11-14 10:56:13", deleted_at: nil, display_on: nil>]
irb(main):014:0> order.distributor.payment_methods                                                                                                                                                                 
=> [#<Spree::PaymentMethod::Check id: 1, type: "Spree::PaymentMethod::Check", name: "Check", description: nil, active: true, environment: "test", created_at: "2018-11-14 10:56:13", updated_at: "2018-11-14 10:56:13", deleted_at: nil, display_on: nil>]                             
irb(main):015:0> order.distributor                                                                                                                                                                                 
=> #<Enterprise id: 1, name: "Enterprise 1", description: "enterprise", long_description: "<p>Hello, world!</p><p>This is a paragraph.</p>", is_primary_producer: nil, contact_name: nil, phone: nil, website: nil, twitter: nil, abn: nil, acn: nil, address_id: 2, pickup_times: nil, next_collection_at: nil, created_at: "2018-11-14 10:55:52", updated_at: "2018-11-14 10:55:52", distributor_info: nil, logo_file_name: nil, logo_content_type: nil, logo_file_size: nil, logo_updated_at: nil, promo_image_file_name: nil, promo_image_content_type: nil, promo_image_file_size: nil, promo_image_updated_at: nil, visible: true, facebook: nil, instagram: nil, linkedin: nil, owner_id: 3, sells: "any", shop_trial_start_date: nil, producer_profile_only: false, permalink: "enterprise-1", charges_sales_tax: false, email_address: nil, require_login: false, allow_guest_orders: true, invoice_text: nil, display_invoice_logo: false, allow_order_changes: false, enable_subscriptions: false>                                                                                  
irb(main):016:0> payment_method.distributors                                                                                                                                                                       
=> [#<Enterprise id: 1, name: "Enterprise 1", description: "enterprise", long_description: "<p>Hello, world!</p><p>This is a paragraph.</p>", is_primary_producer: nil, contact_name: nil, phone: nil, website: nil, twitter: nil, abn: nil, acn: nil, address_id: 2, pickup_times: nil, next_collection_at: nil, created_at: "2018-11-14 10:55:52", updated_at: "2018-11-14 10:55:52", distributor_info: nil, logo_file_name: nil, logo_content_type: nil, logo_file_size: nil, logo_updated_at: nil, promo_image_file_name: nil, promo_image_content_type: nil, promo_image_file_size: nil, promo_image_updated_at: nil, visible: true, facebook: nil, 
instagram: nil, linkedin: nil, owner_id: 3, sells: "any", shop_trial_start_date: nil, producer_profile_only: false, permalink: "enterprise-1", charges_sales_tax: false, email_address: nil, require_login: false, 
allow_guest_orders: true, invoice_text: nil, display_invoice_logo: false, allow_order_changes: false, enable_subscriptions: false>] 
```

Hope all this helps :trollface: